### PR TITLE
lib/connections: Only announce punchable nats (fixes #4519)

### LINKED ISSUE #4519: Avoid lots and lots of announced addresses in the presence of symmetric NAT
When behind symmetric NAT we will get a new port mapping for every STUN lookup, which is (worst case) every 24 seconds. Worst case this means we accumulate up to 150 useless KCP addresses (one hour's worth) on the discovery server.

We should probably just not announce any KCP addresses to the global discovery system when we are behind an unfriendly NAT type.

Relates to #4459

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
Well Tested (TM)

### CODE REVIEW

File: lib/connections/kcp_listen.go [hunk 1]
```diff
@@ -285,11 +292,10 @@ func (t *kcpListener) stunRenewal(listener net.PacketConn) {
...
 				}
 			}
-
-			oldType = natType
 		}
 
-		// We failed to contact all provided stun servers, chillout for a while.
+		// We failed to contact all provided stun servers or the nat is not punchable.
+		// Chillout for a while.
 		time.Sleep(time.Minute)
```
Comment (calmh) [inline lib/connections/kcp_listen.go]: Constantify and raise? a minute is quite a short time isn't it? or maybe that's fine it's just opinion, but constantify at least.

**Comment by calmh** [comment]
@st-review merge

**Comment by st-review** [comment]
@calmh: Build status is `pending`. I'll wait until it goes green and then merge!

**Comment by st-review** [comment]
:ok_hand: Merged as 0518a92cdb025e952a1a405e19306b51e1736595. Thanks, @AudriusButkevicius!

**Pull request closed without merging**